platform/mds/mds-web/src/main/resources/webapp/js/controllers.js

```diff
@@ -4182,15 +4182,15 @@
         * Sets selected instance history by id
         */
         $scope.selectInstanceHistory = function (instanceId) {
-            blockUI();
+            LoadingModal.open();
             History.getHistory({
                 entityId: $scope.selectedEntity.id,
                 instanceId: instanceId
             }, function () {
-                unblockUI();
+                LoadingModal.close();
                 $scope.previousInstance = undefined;
                 $scope.instanceId = instanceId;
-            }, angularHandler('mds.error', 'mds.error.historyRetrievalError'));
+            }, ModalFactory.angularHandler('mds.error', 'mds.error.historyRetrievalError'));
```

Comment:
I still don't see the need for the angularHandler function... I think it might be more clear to handle the error, and then use the familiar ModalFactory.motechAlert() function